src/filter.ts

```diff
@@ -78,7 +78,7 @@ export class Filter {
 
     if (typeof item === 'object') {
       return Object.entries(item).map(([key, pattern]) => {
-        if (typeof key !== 'string' || typeof pattern !== 'string') {
+        if (typeof key !== 'string' || (typeof pattern !== 'string' && !Array.isArray(pattern))) {
```

Comment:
This basically tricks the typescript type system.
`pattern` is declared as `string` but it will be actually array of strings.
Micromatch library also accepts array of string but this signature is not documented there.
It's definitely just a documentation issue. `matcher` method actually delegates to `picomatch()` which indeed accepts string or array of strings.

To keep it simple. Let's leave it as it is and I will do some cleanup afterwards.